server/controllers/customer.js

```diff
@@ -105,5 +105,29 @@ export default {
       throw new ForbiddenError
 
     next()
+  },
+
+  /**
+   * Customer Middleware
+   */
+  async canChangeStatus(req, res, next) {
+    const unrestrictedRoles = [
+      ADMIN_ROLE,
+    ]
+    const restrictedStatus = [
+      customerStatus.REJECTED,
+      customerStatus.PENDING,
+    ]
+
+    if (req.user && !intersection(req.user.roles, unrestrictedRoles).length) {
+      const customer = extend(req.customer, req.body)
+      const oldCustomer = await Customer.findById(customer._id)
```

Comment:
can't you just use `req.customer`?

Reply:
Thanks, I've removed the second lookup in following commit.